### DISCUSSION TITLE
easier side scrolling

### DISCUSSION THREAD

**Discussion by Foxomat** [SpyrexDE/SmoothScroll#47] [Ideas]
when the only way to scroll is sideways, using the mouse wheel (without shift) should scroll sideways (as it is the case with the normal scroll container)

A quick implementation could check if `vertical_scroll_mode` is set to `SCROLL_MODE_DISABLED` in `_gui_input()`, this does not however take into consideration the case when it's not possible to scroll vertically due to lack of scrollable content.

What I added for me ( in `_gui_input()`):
```
	if event is InputEventMouseButton:
		match event.button_index:
			MOUSE_BUTTON_WHEEL_DOWN:
				if event.pressed:
					last_scroll_type = SCROLL_TYPE.WHEEL
					if event.shift_pressed or vertical_scroll_mode == SCROLL_MODE_DISABLED: # <--
						if should_scroll_horizontal():
							velocity.x -= speed
					else:
						if should_scroll_vertical():
							velocity.y -= speed
					friction = friction_scroll
					damping = damping_scroll
			MOUSE_BUTTON_WHEEL_UP:
				if event.pressed:
					last_scroll_type = SCROLL_TYPE.WHEEL
					if event.shift_pressed or vertical_scroll_mode == SCROLL_MODE_DISABLED: # <--
						if should_scroll_horizontal():
							velocity.x += speed
					else:
						if should_scroll_vertical():
							velocity.y += speed
					friction = friction_scroll
					damping = damping_scroll
```

I didn't create a merge request (PR) because maybe you want into more elegantly implemented or also want to include the case with `SCROLL_MODE_AUTO` and not enough scrollable content.

**Comment by SpyrexDE** [comment]
Hey, thanks for pointing that out. You probably overthought it a bit. I could not test it because I am on a laptop without a mouse but it should work just fine by checking `or not should_scroll_vertical()`. :)
